### PR TITLE
Use new template forms for issue templates

### PR DESCRIPTION
**Pull request by flip1995**
GitHub has a new way to define issue templates. This gives an even more consistent formatting of our issues.

You can find (read-only) rendered versions [here](https://github.com/flip1995/rust-clippy/tree/new_templates/.github/ISSUE_TEMPLATE). If you want to try to fill out this kind of issue, you can do this in the [`cargo` Repo](https://github.com/rust-lang/cargo/issues/new/choose).

changelog: Better issue templates

### CODE REVIEW

**Comment by flip1995** [comment]
Of course there is still the question if we want to do this. I'll bring it up in the meeting on Tuesday.

**Comment by xFrednet** [comment]
Very fancy, I like this a lot! :+1:

**Comment by camsteffen** [comment]
I'd like to see error output for false positive and lint name (optional) for new lint. But that's orthogonal to using the fancy new forms, so I think this is good!

File: .github/ISSUE_TEMPLATE/new_lint.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      description: What does this lint do?
+    validations:
+      required: true
+  - type: dropdown
+    id: category
+    attributes:
+      label: Category
+      description: >
+        What category should this lint go into? If you're unsure you can select
+        multiple categories.
```
Comment (camsteffen) [inline .github/ISSUE_TEMPLATE/new_lint.yml:21]: Maybe you could say to see the category descriptions in the readme.

File: .github/ISSUE_TEMPLATE/new_lint.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    attributes:
+      label: Drawbacks
+      description: What might be possible drawbacks of such a lint?
+  - type: textarea
+    id: example
+    attributes:
+      label: Example
+      description: >
+        Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in
+        your environment. E.g. `RUST_BACKTRACE=1 cargo clippy`.
```
Comment (camsteffen) [inline .github/ISSUE_TEMPLATE/new_lint.yml:53]: Copy paste error?

**Comment by flip1995** [comment]
> I'd like to see error output for false positive

I modified the default value of the text area. I think adding an additional section for just an error output would be overkill.

**Review by camsteffen — APPROVED** [review]
r=me after squash. Thanks!

**Comment by flip1995** [comment]
Rebased+Squashed. Let's get this merged.

@bors r=camsteffen

**Pull request merged**